docs/struct.md

```diff
@@ -2,9 +2,170 @@
 
 > "Day one, I'm in love with your struct" Thumpasaurus (kinda)
 
+---
+
+For this chapter, we need to start by activating the necessary feature - in `Cargo.toml`, please make this change:
+
+```diff
+-polars = { version = "0.43.1", default-features = false }
++polars = { version = "0.43.1", features=["dtype-struct"], default-features = false }
+```
+
+---
+
 How do we consume structs, and how do we return them?
 
-To learn about structs, we'll rewrite a plugin which takes a `Struct` as
+Let's try creating a Polars DataFrame in Python that stores a struct similar to this one:
+
+```rust
+struct Point2D {
+    x: f64,
+    y: f64,
+    rgb: u32,
+}
+```
+
+![Diagram showing the struct in a UML-like box](assets/struct_example_Point2D.png){ style="display: block; margin: 0 auto;" }
+
+There are different ways of doing that, but that doesn't matter now. Here's one way:
+```python
+df = pl.DataFrame(
+    {
+        "x": [1.0, 1.25, 1.5, 1.75],
+        "y": [3.0, 2.75, 2.5, 2.25],
+        "rgba": [0x00FF7FFF, 0xFF7F00FF, 0x7F7F7FFF, 0xD8D8D8FF],
+    }
+).select(
+    point_2d_s=pl.struct(
+        "x", "y", "rgba",
+        schema={
+            "x": pl.Float64,
+            "y": pl.Float64,
+            "rgba": pl.UInt32,
+        }
+    )
+)
+```
+
+If we `print(df)`, here's what we have:
+
+```
+shape: (4, 1)
+┌────────────────────────┐
+│ point_2d_s             │
+│ ---                    │
+│ struct[3]              │
+╞════════════════════════╡
+│ {1.0,3.0,16744447}     │
+│ {1.25,2.75,4286513407} │
+│ {1.5,2.5,2139062271}   │
+│ {1.75,2.25,3638089983} │
+└────────────────────────┘
+```
+
+Now's an excellent time to ask: how's that stored in memory? Before we get to that answer, consider this other scenario, in Rust:
+
+```rust
+let v: [Point2D; 4] = [
+    Point2D {
+        x: 1.0,
+        y: 3.0,
+        rgb: 0x00FF7FFFu32,
+    },
+    Point2D {
+        x: 1.25,
+        y: 2.75,
+        rgb: 0xFF7F00FFu32,
+    },
+    Point2D {
+        x: 1.5,
+        y: 2.5,
+        rgb: 0x7F7F7FFFu32,
+    },
+    Point2D {
+        x: 1.75,
+        y: 2.25,
+        rgb: 0xD8D8D8FFu32,
+    },
+];
+```
+
+How's this one stored in memory? You might find that answer easier, it's an array of struct instances, so we have the `x`, `y` and `rgba` fields contiguously in memory, like that:
+
+![Diagram showing the contiguous layout memory of an array of structs in Rust](assets/struct_array_memory_layout.png){ style="display: block; margin: 0 auto;" }
+
+This is consistent with how C, C++, and many other languages store structs in memory.
+How's our struct-in-a-DataFrame different?
+
+Polars follows the Arrow protocol for structs, which means each field of the struct is stored in a Series, backed by chunks. Each chunk is contiguous in memory.
+That makes it so, for instance, assuming a single chunk, the address after 64 bits from the start of the first x field holds the x value for the next struct instance, not the y field of the same instance.
```

Comment:
how about "for the next row" instead of "for the next struct instance"?

Reply:
Got it, will fix it